src/bitcoin-cli.cpp

```diff
@@ -474,9 +532,8 @@ static int CommandLineRPC(int argc, char *argv[])
         }
         std::unique_ptr<BaseRequestHandler> rh;
         std::string method;
-        if (gArgs.GetBoolArg("-getinfo", false)) {
+        if (gArgs.IsArgSet("-getinfo")) {
```

Comment:
Why are you changing this?

Reply:
`IsArgSet()` seemed more appropriate than `GetBoolArg()` as we don't need the value, only to know if the arg is set (and `IsArgSet()` is used elsewhere in this file for the same purpose). Updated the commit message to say why.

Reply:
Pretty sure this will do the wrong thing with `-nogetinfo` or `-getinfo=0` ?

Reply:
Thanks -- with your examples and `IsArgSet`, it runs as if `-getinfo` was passed. With `GetBoolArg`, it raises with "error: too few parameters (need at least command)". If that is the desired behavior, then ISTM I should not only revert this change but also use `GetBoolArg` at line 569 and update the tests to cover this. Confirm?

Reply:
Am also checking for `if (!gArgs.GetArgs("-rpcwallet").empty())` versus `gArgs.IsArgSet("-rpcwallet")`

Reply:
Added a regression test since this was failing silently and reverted to parsing the `-getinfo` and `-rpcwallet` args strictly as before. Thanks @luke-jr for the catch!

```diff
--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -532,7 +532,7 @@ static int CommandLineRPC(int argc, char *argv[])
         }
         std::unique_ptr<BaseRequestHandler> rh;
         std::string method;
-        if (gArgs.IsArgSet("-getinfo")) {
+        if (gArgs.GetBoolArg("-getinfo", false)) {
             rh.reset(new GetinfoRequestHandler());
@@ -543,7 +543,7 @@ static int CommandLineRPC(int argc, char *argv[])
             args.erase(args.begin()); // Remove trailing method name from arguments vector
         }
         Optional<std::string> wallet_name{};
-        if (gArgs.IsArgSet("-rpcwallet")) wallet_name = gArgs.GetArg("-rpcwallet", "");
+        if (!gArgs.GetArgs("-rpcwallet").empty()) wallet_name = gArgs.GetArg("-rpcwallet", "");
         const UniValue reply = ConnectAndCallRPC(rh.get(), method, args, wallet_name);
@@ -566,7 +566,7 @@ static int CommandLineRPC(int argc, char *argv[])
         } else {
-            if (gArgs.IsArgSet("-getinfo") && !gArgs.IsArgSet("-rpcwallet")) {
+            if (gArgs.GetBoolArg("-getinfo", false) && gArgs.GetArgs("-rpcwallet").empty()) {
                 GetWalletBalances(result); // fetch multiwallet balances and append to result
             }
diff --git a/test/functional/interface_bitcoin_cli.py b/test/functional/interface_bitcoin_cli.py
index 7530e7daf6..10ac82b107 100755
--- a/test/functional/interface_bitcoin_cli.py
+++ b/test/functional/interface_bitcoin_cli.py
@@ -49,6 +49,10 @@ class TestBitcoinCli(BitcoinTestFramework):
         self.log.info("Test -getinfo with arguments fails")
         assert_raises_process_error(1, "-getinfo takes no arguments", self.nodes[0].cli('-getinfo').help)
 
+        self.log.info("Test -getinfo=0 and -nogetinfo fail")
+        for command in ['-getinfo=0', '-nogetinfo']:
+            assert_raises_process_error(1, "error: too few parameters (need at least command)", self.nodes[0].cli(command).send_cli)
+
         self.log.info("Test -getinfo returns expected network and blockchain info")
```

Reply:
Seems fine, although this is a pretty pathological case. Why would anyone ever call `bitcoin-cli -nogetinfo`?

Reply:
Fix and test coverage added in #19089